File: ruby/resistor-color-duo/resistor_color_duo.rb
class ResistorColorDuo
  COLORS = {
    black: 0,
    brown: 1,
    red: 2,
    orange: 3,
    yellow: 4,
    green: 5,
    blue: 6,
    violet: 7,
    grey: 8,
    white: 9,
  }

  def self.value(arr)
    arr.first(2)
       .map { |i| COLORS[i.to_sym] }
       .join
       .to_i
  end
end

File: typescript/resistor-color-duo/resistor-color-duo.ts
export class ResistorColor {
  private colors: string[];
  
  COLORS = [
    "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
  ]

  constructor(colors: string[]) {
    if (colors.length < 2) { 
      const message = "At least two colors need to be present"
      throw new Error(message)
    }

    this.colors = colors;
  }
  
  value = (): number => {
    return +(this.colors.slice(0, 2).map(color => this.COLORS.indexOf(color)).join(""))
  }
}
